tests/providers/amazon/aws/operators/test_base_aws.py

```diff
@@ -0,0 +1,132 @@
+# under the License.
+from __future__ import annotations
+
+import pytest
+
+from airflow.exceptions import AirflowException, AirflowProviderDeprecationWarning
+from airflow.hooks.base import BaseHook
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.providers.amazon.aws.operators.base_aws import AwsBaseOperator
+
+TEST_CONN = "aws_test_conn"
+
+
+class FakeS3Hook(AwsBaseHook):
+    """Hook for tests, implements thin-wrapper around s3 client."""
+
+    def __init__(self, **kwargs):
+        kwargs.update({"client_type": "s3", "resource_type": None})
+        super().__init__(**kwargs)
+
+
+class FakeS3Operator(AwsBaseOperator):
+    aws_hook_class = FakeS3Hook
+
+
+@pytest.fixture(autouse=True)
+def fake_conn(monkeypatch):
+    monkeypatch.setenv(f"AWS_CONN_{TEST_CONN.upper()}", '{"conn_type": "aws"}')
+
+
+class TestAwsBaseOperator:
+    def test_default_parameters(self):
+        op = FakeS3Operator(task_id="fake_task_id")
+        msg = "Attention! Changes in default parameters might produce breaking changes in multiple operators"
+        assert op.aws_conn_id == "aws_default", msg
+        assert op.region_name is None, msg
+        assert op.verify is None, msg
+        assert op.botocore_config is None, msg
+
+    def test_parameters(self):
+        op = FakeS3Operator(
+            task_id="fake-task-id",
+            aws_conn_id=TEST_CONN,
+            region_name="eu-central-1",
+            verify=False,
+            botocore_config={"read_timeout": 777, "connect_timeout": 42},
+        )
+
+        assert op.aws_conn_id == TEST_CONN
+        assert op.region_name == "eu-central-1"
+        assert op.verify is False
+        assert op.botocore_config == {"read_timeout": 777, "connect_timeout": 42}
+
+        hook = op.hook
+        assert isinstance(hook, FakeS3Hook)
+        assert hook.aws_conn_id == op.aws_conn_id
+        assert hook._region_name == op.region_name
+        assert hook._verify == op.verify
+        assert hook._config.read_timeout == 777
+        assert hook._config.connect_timeout == 42
+
+    @pytest.mark.parametrize(
+        "region, region_name",
+        [
+            pytest.param("eu-west-1", None, id="region-only"),
+            pytest.param("us-east-1", "us-east-1", id="non-ambiguous-params"),
+        ],
+    )
+    def test_deprecated_region_name(self, region, region_name):
+        warning_match = r"`region` is deprecated and will be removed"
+        with pytest.warns(AirflowProviderDeprecationWarning, match=warning_match):
+            op = FakeS3Operator(
+                task_id="fake-task-id",
+                aws_conn_id=TEST_CONN,
+                region=region,
+                region_name=region_name,
+            )
+        assert op.region_name == region
+
+        with pytest.warns(AirflowProviderDeprecationWarning, match=warning_match):
+            assert op.region == region
+
+    def test_ambiguous_region_name(self):
```

Comment:
If you accept the suggestion to reword the warning, you'll need to change this as well.